common/springUtilities/unitScripts.lua

```diff
@@ -0,0 +1,420 @@
+	end
+
+	-- Check for multiline opening "long bracket" [=*[
+	local function multilineOpenLength(pos)
+		local j = pos + 1 -- pos is index of '['
+		local count = 0
+		while j <= n and chartAt(j) == "=" do
+			count = count + 1
+			j = j + 1
+		end
+		if j <= n and chartAt(j) == "[" then
+			return count
+		end
+	end
+
+	-- Skip closing ]=] of given size
+	local function skipMultilineClose(count, start_pos)
+		local j = start_pos
+		while j <= n do
+			if chartAt(j) == "]" then
+				local k = j + 1
+				local eq2 = 0
+				while k <= n and chartAt(k) == "=" do
+					eq2 = eq2 + 1
+					k = k + 1
+				end
+				if k <= n and chartAt(k) == "]" and eq2 == count then
+					return k + 1
+				end
+				j = j + 1
+			else
+				j = j + 1
+			end
+		end
+		return nil
+	end
+
+	while i <= n do
+		local ch = current()
+
+		if quotes[ch] then
+			-- Literals
+			local quote = ch
+			insert(output, quote)
+			i = i + 1
+			while i <= n do
+				local c = current()
+				insert(output, c)
+				i = i + 1
+				if c == escape then
+					if i <= n then
+						insert(output, current())
+						i = i + 1
+					end
+				elseif c == quote then
+					break
+				end
+			end
+
+		elseif starts_with("--") then
+			i = i + 2
+
+			if i <= n and current() == "[" then
+				local length = multilineOpenLength(i)
+
+				if length then
+					-- Multiline comment
+					i = i + 1 + length + 1
+					local next_pos = skipMultilineClose(length, i)
+
+					if next_pos then
+						local k = i
+						while k < next_pos do
+							local c = chartAt(k)
+							if line_ends[c] then insert(output, c) end
+							k = k + 1
+						end
+						i = next_pos
+					else
+						while i <= n do
+							local c = current()
+							if line_ends[c] then insert(output, c) end
+							i = i + 1
+						end
+					end
+				else
+					-- Single-line comment
+					while i <= n do
+						local c = current()
+						i = i + 1
+						if c == line_return then
+							insert(output, line_return)
+							if i <= n and current() == line_feed then
+								insert(output, line_feed)
+								i = i + 1
+							end
+							break
+						elseif c == line_feed then
+							insert(output, line_feed)
+							break
+						end
+					end
+				end
+			else
+				-- Single-line comment
+				while i <= n do
+					local c = current()
+					i = i + 1
+					if c == line_return then
+						insert(output, line_return)
+						if i <= n and current() == line_feed then
+							insert(output, line_feed)
+							i = i + 1
+						end
+						break
+					elseif c == line_feed then
+						insert(output, line_feed)
+						break
+					end
+				end
+			end
+
+		else
+			insert(output, ch)
+			i = i + 1
+		end
+	end
+
+	return table.concat(output)
+end
+
+---Test whether a unit's script is COB, as opposed to LUS.
+---@param unitDef table
+---@return boolean
+local function isUnitScriptCOB(unitDef)
+	local scriptName = unitDef.scriptName
+	return scriptName and scriptName:lower():find(".cob$") and true or false
```

Comment:
i think extensionless files are also treated as cob by default (i.e. `script = "foo"` might try to load "foo" as a cob file, or maybe even "foo.cob")